lib/polymorphic_integer_type/extensions.rb

```diff
@@ -45,6 +46,7 @@ def belongs_to(name, scope = nil, **options)
             define_method "#{name}=" do |record|
               super(record)
               send("#{foreign_type}=", record.class)
+              association(name).loaded!
```

Comment:
What's the reason behind this change?

Reply:
if we write the association in the `super(record)` and then change then `*_type` column, it will mark the association as stale, and lose the reference of it (causing it to want to lookup the object again). 

By marking the assocaition as loaded afterwards, we guarantee that if once you write the record, it maintains that association.